state_manage/reducers.ts

```diff
@@ -0,0 +1,35 @@
+import * as ActionTypes from './actions/actionTypes';
+import { combineReducers } from 'redux';
+
+const initialState = {
+    profile: {
+        'name': 'Sergio Diaz',
+        'bio': 'Junior in the college of engineering. Selling a bunch of textbooks and clothes I don\'t need.'
+    }
+}
+
+const profileReducer = (state = initialState, action) => {
+    console.log(state)
+    switch (action.type) {
+        case ActionTypes.SET_NAME:
+            return {
+                ...state,
+                'name': action.name
```

Comment:
is this correct? `'name'` is under `profile`, so don't you need to do something like this:
```
{ profile } = state;
newProfile = { ...profile, 'name': action.name};
return {
      ...state,
      profile: newProfile
}
```

Reply:
That actually duplicates profile within state, so the state looks something like this:
`
object {
  profile {
    profile {
      name: ...
      bio: ...
    }
  }
}
`

Reply:
Not sure why, but that's actually what the print statement I forgot to get rid of was for. Redux wasn't working, and I found out the reducer was causing a duplicate profile object to be created

Reply:
I still don't think your reducer is correct. I tried calling the reducer with action `ActionTypes.SET_NAME` and payload 'Haichen Wang' but it returns 
```
{
   name: "Haichen Wang", 
   profile: {
          name: 'Sergio Diaz',
          bio: 'Junior in the college of engineering. Selling a bunch of textbooks and clothes I don\'t need.'
    }
}
```

It is a good practice to implement tests, especially for redux. I have made a commit to your branch for setting up tests. We're using Jest to do testing. If you pull you'll find a directory `__tests__` under your `state_manage` folder, and `reducers.test.js` contains the test I described above. The test can be run with command `npm run test`, which shows that the test fails:
<img width="808" alt="Screen Shot 2021-12-27 at 3 48 19 PM" src="https://user-images.githubusercontent.com/55666780/147448367-13752d72-7b18-431b-ace7-ef688bd3fb19.png">

If you have time, it'd be great if you also implement tests for other actions used by your reducer. You can reference the test I wrote and [Jest documentation](https://jestjs.io/docs/getting-started#using-babel).

Reply:
Even though your test case failed, the dummy example I put on the profile screen (you can see in the video) still worked. Once I got my code to pass your test case, the dummy example started to fail. Not sure why this is, maybe I made a mistake on profile page I'll check it out now.